app/pages/EditBreakingNewsPage.jsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { RiAddBoxLine, RiDeleteBin5Line, RiSave3Line } from 'react-icons/ri';
+import { withRouter } from 'react-router-dom';
+
+import * as dataService from '../utils/DataService';
+import './EditBreakingNewsPage.scss';
+
+class EditBreakingNewsPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { breakingNewsItems: [] };
+
+    this.onAddNew = this.onAddNew.bind(this);
+    this.onDelete = this.onDelete.bind(this);
+  }
+
+  async componentDidMount() {
+    const { news_articles } = await dataService.get('/api/news_articles');
+    const breakingNewsItems = news_articles.map(article => ({
+      ...article,
+      effective_date: this.formatDate(article.effective_date),
+      expiration_date: this.formatDate(article.expiration_date)
+    }));
+
+    this.setState({ breakingNewsItems });
+  }
+
+  formatDate(date) {
+    return date ? new Date(date) : null;
+  }
+
+  onFieldChange(item, { target }) {
+    const { value, dataset } = target;
+    const { field } = dataset;
+    const newItem = { ...item, [field]: value };
+
+    this.setState(prevState => ({
+      breakingNewsItems: prevState.breakingNewsItems
+        .map(oldItem => (oldItem.id === newItem.id ? newItem : oldItem)),
+    }));
+  }
+
+  onDatePickerChange(item, value, field) {
+    const newItem = { ...item, [field]: value };
+
+    this.setState(prevState => ({
+      breakingNewsItems: prevState.breakingNewsItems
+        .map(oldItem => (oldItem.id === newItem.id ? newItem : oldItem)),
+    }));
+  }
+
+  onSave(item) {
+    dataService.put(`/api/news_articles/${item.id}`, item);
+  }
+
+  async onDelete(item) {
+    await dataService.APIDelete(`/api/news_articles/${item.id}`);
+
+    this.setState(prevState => ({
+      breakingNewsItems: prevState.breakingNewsItems
+        .filter(oldItem => oldItem.id !== item.id),
+    }));
+  }
+
+  async onAddNew() {
+    const response = await dataService.post('/api/news_articles', {});
```

Comment:
I'm a little bit surprised that the API works this way, since I think it complicates things when we don't do any validation or even have a database column constraint that requires that any of the fields be set.

@jjfreund, did we specifically design the API such that none of the fields is mandatory, and if not, could we add some validation both at the Rails controller layer and the DB schema layer to require that these fields not be null?

I'm also noticing some unintuitive behavior on the edit page where if I click "Add Breaking News Story" and refresh before actually populating the fields or saving, then the empty item disappears from the UI. It looks like the empty items still get created in the database, but they never get displayed because the `NewsArticlesController#index` method apparently filters out any articles that have a null `effective_date`. Basically, if you fail to set the "effective date" and save the article before refreshing the page, it gets lost to the DB and cannot be surfaced in the edit UI.

I don't know if this was intentional, since we [only do an explicit null check for `expiration_date`, not `effective_date`](https://github.com/ShelterTechSF/askdarcel-api/blob/6a64dfa11d732e8c880f298e634ef4d24c355ac1/app/controllers/news_articles_controller.rb#L13), and I guess SQL treats `effective_date < ?` as "false" if `effective_date` is null.